backend/src/main/java/com/redhat/cloud/notifications/routers/internal/errata/ErrataUserPreferencesMigrationResource.java

```diff
@@ -0,0 +1,58 @@
+package com.redhat.cloud.notifications.routers.internal.errata;
+
+import com.redhat.cloud.notifications.Constants;
+import com.redhat.cloud.notifications.auth.ConsoleIdentityProvider;
+import io.quarkus.logging.Log;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.inject.Inject;
+import jakarta.validation.constraints.NotNull;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.MediaType;
+import org.jboss.resteasy.reactive.RestForm;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+@Path(Constants.API_INTERNAL + "/team-nado")
+@RolesAllowed(ConsoleIdentityProvider.RBAC_INTERNAL_ADMIN)
+public class ErrataUserPreferencesMigrationResource {
+    private static final String USERNAME_KEY = "username";
+    private static final String ORG_ID_KEY = "org_id";
+    private static final String[] CSV_HEADERS = {USERNAME_KEY, ORG_ID_KEY};
+
+    @Inject
+    ErrataMigrationRepository errataMigrationRepository;
+
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Path("/migrate/json")
+    @POST
+    public void migrateTeamNadoUserPreferencesJSON(@NotNull @RestForm("jsonFile") InputStream jsonFile) throws IOException {
+        final byte[] contents = jsonFile.readAllBytes();
+
+        // Extract the subscriptions from the JSON file.
+        final String stringContents = new String(contents, StandardCharsets.UTF_8);
+        final JsonArray jsonArray = new JsonArray(stringContents);
+
+        final List<ErrataSubscription> errataSubscriptions = new ArrayList<>();
+        for (final Object rawJsonObject : jsonArray) {
+            final JsonObject teamNadoSubscription = (JsonObject) rawJsonObject;
+
+            errataSubscriptions.add(new ErrataSubscription(teamNadoSubscription.getString(USERNAME_KEY), teamNadoSubscription.getString(ORG_ID_KEY)));
```

Comment:
according https://issues.redhat.com/browse/RHCLOUD-30812, the current portal allow user to select the equivalent of or event type, I think we must honor those choices.

Reply:
Done too!